src/AWS/Orleans.Reminders.DynamoDB/Reminders/DynamoDBReminderTable.cs

```diff
@@ -146,31 +146,48 @@ public async Task<ReminderTableData> ReadRows(GrainReference grainRef)
         /// <summary>
         /// Reads reminder table data for a given hash range.
         /// </summary>
-        /// <param name="beginHash"></param>
-        /// <param name="endHash"></param>
+        /// <param name="begin"></param>
+        /// <param name="end"></param>
         /// <returns> Return the RemiderTableData if the rows were read successfully </returns>
-        public async Task<ReminderTableData> ReadRows(uint beginHash, uint endHash)
+        public async Task<ReminderTableData> ReadRows(uint begin, uint end)
         {
-            var expressionValues = new Dictionary<string, AttributeValue>
-                {
-                    { $":{SERVICE_ID_PROPERTY_NAME}", new AttributeValue(this.serviceId) },
-                    { $":Begin{GRAIN_HASH_PROPERTY_NAME}", new AttributeValue { N = beginHash.ToString() } },
-                    { $":End{GRAIN_HASH_PROPERTY_NAME}", new AttributeValue { N = endHash.ToString() } }
-                };
+            Dictionary<string, AttributeValue> expressionValues = null;
 
             try
             {
                 string expression = string.Empty;
-                if (beginHash < endHash)
+                List<ReminderEntry> records;
+
+                if (begin < end)
                 {
-                    expression = $"{SERVICE_ID_PROPERTY_NAME} = :{SERVICE_ID_PROPERTY_NAME} AND {GRAIN_HASH_PROPERTY_NAME} > :Begin{GRAIN_HASH_PROPERTY_NAME} AND {GRAIN_HASH_PROPERTY_NAME} <= :End{GRAIN_HASH_PROPERTY_NAME}";
+                    expressionValues = new Dictionary<string, AttributeValue>
+                    {
+                        { $":{SERVICE_ID_PROPERTY_NAME}", new AttributeValue(this.serviceId) },
+                        { $":Begin{GRAIN_HASH_PROPERTY_NAME}", new AttributeValue { N = begin.ToString() } },
+                        { $":End{GRAIN_HASH_PROPERTY_NAME}", new AttributeValue { N = end.ToString() } }
+                    };
+                    expression = $"{SERVICE_ID_PROPERTY_NAME} = :{SERVICE_ID_PROPERTY_NAME} AND {GRAIN_HASH_PROPERTY_NAME} BETWEEN :Begin{GRAIN_HASH_PROPERTY_NAME} AND :End{GRAIN_HASH_PROPERTY_NAME}";
```

Comment:
Please correct me if I'm wrong, but I believe this changes the query semantics. It's supposed to be `> Begin && <= End` (non-inclusive lower bound, inclusive upper), but `BETWEEN` is equivalent to `=> Begin && <= End` (inclusive bounds)

Reply:
I think you're correct.  Since performing a dynamodb query with an AND is not possible, would it be sufficient to pas the end attribute value as end-1 and still use BETWEEN?

Reply:
@ReubenBond is correct. The query semantics are changed. I've overlooked this. The original one was correct. Are you sure this passed the unit tests? I remember the reminder tests being very nitpick about this.

Reply:
I missed an unrelated change in DynamoDBStorage that has now been corrected.
AWUtils.Tests.RemindersTest passes with the questionable code as-is, and also if I change line 167 to (end - 1).ToString(), which I believe is the correct change to make.

Reply:
Shouldn't it be `begin + 1`, rather than `end - 1`? The lower bound is supposed to be exclusive, not the upper bound.
It will work but please wrap the expression in `unchecked(...)` , since it's possible for it to overflow just to make it extra obvious that it's expected behavior

Reply:
yes, thank you - vacation brain hasn't recovered yet.

Reply:
No sweat, I'm in the same boat

Reply:
Of course... it's impossible for both `begin < end` and `begin == uint.MaxValue` to be true at the same time, so the `unchecked` is entirely meaningless as no overflow is possible 😅